src/pocketbase/services/batch.py

```diff
@@ -0,0 +1,143 @@
+import json
+from typing import TYPE_CHECKING, Any, Literal, TypedDict
+from urllib.parse import quote
+
+from pocketbase.models.options import CommonOptions, SendOptions
+from pocketbase.services.base import Service
+from pocketbase.utils.types import BodyDict, JsonType, SendableFiles, transform
+
+if TYPE_CHECKING:
+    from pocketbase.client import PocketBase, PocketBaseInners
+
+
+class BatchRequest(TypedDict, total=False):
+    url: str
+    method: Literal["GET", "POST", "PUT", "PATCH", "DELETE"]
+    headers: dict[str, str] | None
+    body: BodyDict
+
+
+class BatchRequestResult(TypedDict, total=False):
+    status: int
+    body: BodyDict
+
+
+class BatchService(Service):
+    __base_sub_path__: str = "/api/batch"
+    """
+    Service for batching multiple PocketBase API requests into a single `/api/batch` call.
+    """
+
+    def __init__(self, pocketbase: "PocketBase", inners: "PocketBaseInners") -> None:
+        super().__init__(pocketbase, inners)
+        self._requests: list[BatchRequest] = []
+        self._subs: dict[str, SubBatchService] = {}
+
+    # we can use it with `async with` syntax
+    async def __aenter__(self) -> "BatchService":
+        return self
+
+    async def __aexit__(
+        self,
+        exc_type: type[BaseException] | None,
+        exc: BaseException | None,
+        tb: Any,
+    ) -> None:
+        return None
```

Comment:
The point is to collect requests/subs within this enter/exit scope (and not allow it outside of that range) and then call await self._send() within the scope exit (if no exception is set). This means the user does not need to call send at all.